### DISCUSSION TITLE
Has this code been abandoned?

### DISCUSSION THREAD

**Discussion by joeinaustin** [taligentx/dscKeybusInterface#332] [dscKeybusInterface Discussion]
I just found this library and am very interested in using it.  However, most of the issues are years old and unanswered.  Is anyone still interested in using/updating/maintaining it??

**Comment by kghandi** [comment]
I have been using this library for several years without issues as I am sure have many others. Your success and whether any of the bug reports are relevant to you will depend on the type of alarm system you have.

**Reply by joeinaustin** [reply]
Thanks for the reply.  I am glad to hear that some have had success.  I immediatly ran into an issue just trying to load the first example.  The arduino IDE says the code is too large for the Uno board.  This despite the docs saying the Uno is supported.  So far the only Arduino that seems to have enough memory is the Mega.  What board and IDE are you using??

**Reply by kghandi** [reply]
I am using an ESP8266 (ESP-12F) board, more specifically a WeMos D1 Mini (Amazon: https://a.co/d/80Zavly). I added a little DC-DC power regulator (Amazon: https://a.co/d/hARP1g1) to supply power from my alarm control panel (which is a DSC Power832 PC5010) and a couple of resistors and a transistor for level shifting as was suggested in the Readme notes. I am using the HomeAssistant-MQTT example code which I only modified to set my Wifi & Mqtt credentials. It has been working since I set it up in 2021, and I haven't kept up with any updates to the library since.

**Reply by kricon** [reply]
Hi @joeinaustin and sorry for late answer. Same as others, I'm using this example for years with various examples without any troubles... I'm also using 4N35 optocouplers just to be sure and isolate it from panel.

**Issue you're having with KeypadReader example shetch outgrew Arduino memory is fixed in latest [develop branch](https://github.com/taligentx/dscKeybusInterface/tree/develop).** 
You need to download it and manually place into \Documents\Arduino\libraries folder.
Other examples works just fine without memory issues.

As for maintenance/update, AFAIK there are some not-mandatory features planned but as life thing goes on there is less free time to implement it. And the main point is - for most people out there library works just fine, for others is mostly wiring/panel not supported problem.

**Comment by Dilbert66** [comment]
As features get added and code gets updated, trying to keep code small enough to work on limited hardware becomes a fruitless task and unfortunately this is the case now. I would recommend  an ESP8266 or better yet an ESP32

**Comment by frankonski** [comment]
> Is anyone still interested in using/updating/maintaining it??

The fact that this library is licensed under GPLv3 makes it one to avoid using in any product you’d want to distribute… so aside from personal use, maybe that is why contributions are small.
